server.js

```diff
@@ -1,38 +1,161 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixData from './data/netflix-titles.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const movieSchema = new mongoose.Schema({
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String, 
+  date_added: String,
+  release_year: Number,
+  rating: String, 
+  duration: String, 
+  listed_in: String,
+  description: String, 
+  type: String
+})
+
+const Movie = mongoose.model('Movie', movieSchema)
+
+// Function to start seeding our Database
+// will only run if RESET_DB environment variable is present and is true
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    // Starts by deleting any pre-existing Movie objects to prevent duplicates when running database
+    await Movie.deleteMany()
+
+    netflixData.forEach(async (show) => {
+      const newMovie = new Movie(show)
+      await newMovie.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/movies', async (req, res) => {
+  const { country, cast, show, page, per_page } = req.query
+
+  await Movie.updateMany(
+    { }, 
+    { 
+      $unset: { show_id: "" }
+    }
+  )
+  const movies = await Movie.aggregate([
+    { 
+      $match: {
+        country: {
+          $regex: new RegExp(country, "i")
+        }, 
+        cast: {
+          $regex: new RegExp(cast, "i")
+        },
+        type: {
+          $regex: new RegExp(show, "i")
+        }
+      }
+    }, 
+    {
+      $skip: Number((page + 1) * per_page + 1)
+    }/* , 
+    {
+      $limit: Number(per_page)
```

Comment:
you need to provide the default `per_page` value because here you trying parsing undefined into the number if your user will not provide you with value in the query. 

```terminal
> Number(undefined)
< NaN
```

Reply:
When something is yealing about value, remember to console.log value the input before and after the function. 

```js
console.log(typeof(per_page))
const numberPefPage = Number(per_page)
console.log(typeof(numberPerPage))
```

and then you would see 
```
< undefined
< NaN
```